### PR TITLE
chore: Make perf test use out/replicache

### PR DESCRIPTION
**Pull request by arv**
This is so that we are testing something closer to what we are building.

### CODE REVIEW

File: src/replicache-options.ts [hunk 1]
```diff
@@ -224,4 +224,9 @@ export type ReplicacheInternalOptions = {
    * Defaults to true.
    */
   enableMutationRecovery?: boolean;
+
+  /**
+   * Allows exposing _persist to perf test
+   */
+  exposePersistMethodAs?: symbol;
```
Comment (arv) [inline src/replicache-options.ts:231]: @grgbkr This is another new "hack" to expose internals.

When we minimize we rename all properties that start with `_` so `_persist` gets renamed.

Instead we expose the method on this using a symbol which the subclass passes in.

File: rollup.config.js [hunk 2]
```diff
@@ -6,13 +6,16 @@ import dts from 'rollup-plugin-dts';
 // We use esbuild for building the actual code.
 
 export default {
-  input: 'out/.dts/src/mod.d.ts',
+  input: 'out/.dts/mod.d.ts',
```
Comment (arv) [inline rollup.config.js:9]: argh...

The output uses the shortest common sub path and now that perf/ is no longer part of the project that changed.

File: src/replicache-options.ts (hunk 1 above)
Comment (grgbkr) [reply src/replicache-options.ts:231]: I don't love it.  I don't hate it.  I don't have a better suggestion.

File: perf/replicache.ts [hunk 3]
```diff
@@ -62,12 +62,18 @@ export function benchmarkPopulate(opts: {
...
   constructor(options: ReplicacheOptions<MD>) {
-    super(options);
+    super({
+      ...options,
+      exposePersistMethodAs: persistSymbol,
+    } as ReplicacheOptions<MD>);
   }
   async persist(): Promise<void> {
-    return this._persist();
+    return this[persistSymbol]();
```
Comment (grgbkr) [inline perf/replicache.ts:76]: surprised this type checks.

File: tool/build.mjs [hunk 4]
```diff
@@ -49,27 +52,18 @@ async function buildCLI() {
...
-    outfile: 'perf/index.js',
-    entryPoints: ['perf/index.ts'],
-    define: {
-      'process.env.NODE_ENV': '"production"',
-    },
-    sourcemap: true,
-    format: 'esm',
-    minify: true,
-  });
+  // Same as what wev build for the npm package but we turn on sourcemaps.
```
Comment (grgbkr) [inline tool/build.mjs:55]: I think we should also always publish a private package of replicache with source maps.  It's hard to debug issues in our sample apps (that we suspect may be a replicache bug) without source maps.

File: src/replicache-options.ts (hunk 1 above)
Comment (grgbkr) [reply src/replicache-options.ts:231]: Something that may type check a bit better, and also can more easily expose additional internal apis.
```
interface ReplicacheInternalOptions {
  receiveInternalAPI?: (internalAPI: ReplicacheInternalAPI) => void;
}

interface ReplicacheInternalAPI {
  persist(): Promise<void>;
}
```

**Review by grgbkr — COMMENTED** [review]
LGTM

File: tool/build.mjs (hunk 4 above)
Comment (arv) [reply tool/build.mjs:55]: That is a good idea... Or we could publish a private package that is not minimized? But sourcemaps might be a better start.

**Pull request merged**